### PR TITLE
[WIP] Drop cloud provider specific bits of cadvisor

### PR DESCRIPTION
**Pull request by dims**
A very long time age in-tree cloud providers depended on cadvisor to pull `InstanceType` and `InstanceID`. The in-tree cloud providers we have left in kubernetes/kubernetes do not depend or need these fields from cadvisor[1] anymore. This is a big vector of dragging in external dependencies, so we should clean this up.

[1] https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/google/cadvisor/machine/info.go#L122-L123

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
/assign @derekwaynecarr @yujuhong 

cc @andrewsykim @cheftako

**Review by mattjmcnaughton — APPROVED** [review]
Fantastic!

/lgtm
/test pull-kubernetes-conformance-kind-ipv6-parallel

**Comment by andrewsykim** [comment]
/lgtm

as well, good catch @dims

File: pkg/kubelet/cadvisor/cadvisor_linux.go [hunk 1]
```diff
@@ -32,12 +32,6 @@ import (
 	_ "github.com/google/cadvisor/container/docker/install"
 	_ "github.com/google/cadvisor/container/systemd/install"
 
-	// Register cloud info providers.
-	// TODO(#76660): Remove this once the cAdvisor endpoints are removed.
```
Comment (dashpole) [inline pkg/kubelet/cadvisor/cadvisor_linux.go]: This says they should be removed after the endpoints are removed.  Should we wait until that is the case?  I believe they were just disabled this release.

Comment (dims) [reply pkg/kubelet/cadvisor/cadvisor_linux.go]: we should ask @tallclair ( https://github.com/kubernetes/kubernetes/pull/76912 )

Comment (derekwaynecarr) [reply pkg/kubelet/cadvisor/cadvisor_linux.go]: i would like confirmation from others as well before removing.

**Comment by dims** [comment]
/assign @tallclair 

@tallclair , Can you please chime in? (based on your comments from https://github.com/kubernetes/kubernetes/pull/76912)

File: pkg/kubelet/cadvisor/cadvisor_linux.go (hunk 1 above)
Comment (tallclair) [reply pkg/kubelet/cadvisor/cadvisor_linux.go]: The motivation for waiting until the endpoints were removed was that this is a breaking change. By waiting for the cAdvisor endpoint removal (which we want to do anyway) I was deferring the work of making the breaking change to cAdvisor.

So, the first question is, what is the status of the removal of the cAdvisor endpoints? Does it have a date? If not, what's blocking it? I think cAdvisor is not yet the longpole for cloud provider removal, so if the planned cAdvisor endpoint removal date is before the final cloud provider removal date we could differ this.

If we definitely want to remove this before the cAdvisor endpoints are removed, we should probably treat this as a breaking change and take the usual precautions & process. I don't know of anyone using the cAdvisor endpoints, but I don't think we have a way of confirming that.

**Pull request closed without merging**